### PR TITLE
refactor(table): remove HOC

### PR DESCRIPTION
**Pull request by montezume**
#### Summary

Refactors the `Table` component to remove usage of `withMouseOverState`

### CODE REVIEW

File: src/components/table/sortable-header/sortable-header.js [hunk 1]
```diff
@@ -27,43 +27,59 @@ Logic of arrow indicating sort order is slightly complex:
...
+const Container = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+
+  &:hover {
+    cursor: pointer;
+
+    svg {
+      transform: rotate(180deg);
```
Comment (montezume) [inline src/components/table/sortable-header/sortable-header.js]: 😉

File: src/components/table/sortable-header/sortable-header.js [hunk 2]
```diff
@@ -74,12 +90,12 @@ const SortableHeader = props => {
         ]}
       >
         {isArrowDown ? (
-          <AngleDownIcon size="medium" theme={theme} />
+          <AngleDownIcon size="medium" theme="white" />
         ) : (
-          <AngleUpIcon size="small" theme={theme} />
```
Comment (montezume) [inline src/components/table/sortable-header/sortable-header.js]: I'm pretty sure this was a bug... It looked bad.

File: src/components/table/sortable-header/sortable-header.js (hunk 1 above)
Comment (emmenko) [reply src/components/table/sortable-header/sortable-header.js]: Cheater! 😜

**Comment by emmenko** [comment]
What's wrong with percy?

**Comment by montezume** [comment]
@emmenko I think it just didn't get approved right from the last time. I'll approve it again.

**Pull request merged**